staging/src/k8s.io/cli-runtime/pkg/resource/fallback_query_param_verifier.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/openapi"
+	"k8s.io/klog/v2"
+)
+
+// fallbackQueryParamVerifier encapsulates a OpenAPI V3 and and
+// an OpenAPI V2 query param verifier.
+type fallbackQueryParamVerifier struct {
+	v3Verifier Verifier
+	v2Verifier Verifier
+}
+
+var _ Verifier = &fallbackQueryParamVerifier{}
+
+// NewFallbackQueryParamVerifier creates both the V3 and V2 openAPI
+// query param verifiers, storing them in the fallback struct.
+func NewFallbackQueryParamVerifier(
+	dynamicClient dynamic.Interface,
+	oapiClient openapi.Client,
+	openAPIGetter discovery.OpenAPISchemaInterface,
+	queryParam VerifiableQueryParam) Verifier {
+
+	// Create the v3 and v2 QueryParamVerifiers.
+	v3Verifier := NewQueryParamVerifierV3(dynamicClient, oapiClient, queryParam)
+	v2Verifier := NewQueryParamVerifier(dynamicClient, openAPIGetter, queryParam)
+	return &fallbackQueryParamVerifier{
+		v3Verifier: v3Verifier,
+		v2Verifier: v2Verifier,
+	}
+}
```

Comment:
Not a strong opinion, we could have built these out of that constructor and do ...
```suggestion
func NewFallbackQueryParamVerifier(primary, secondary Verifier) Verifier {
	return &fallbackQueryParamVerifier{
		primary: primary,
		secondary: secondary,
	}
}
```

Reply:
I like this, and I've changed to this approach. I've also added unit tests.